src/engraving/automation/iautomation.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-Studio-CLA-applies
+ *
+ * MuseScore Studio
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2025 MuseScore Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "automationtypes.h"
+
+#include "global/types/bytearray.h"
+#include "global/types/id.h"
+
+namespace mu::engraving {
+class IAutomation
+{
+public:
+    virtual ~IAutomation() = default;
+
+    virtual const AutomationCurve& curve(const muse::ID& staffId) const = 0;
+
+    virtual void addPoint(const muse::ID& staffId, int utick, const AutomationPoint& p) = 0;
+    virtual void removePoint(const muse::ID& staffId, int utick) = 0;
+    virtual void setPointValue(const muse::ID& staffId, int utick, double value) = 0;
```

Comment:
In order to move a point horizontally, do we have to delete it at the old tick and insert it at the new tick?